test/metabase/api/card_test.clj

```diff
@@ -189,6 +189,46 @@
                                 :target [:variable [:template-tag :category]]
                                 :value  2}]})))))))
 
+(deftest run-query-with-default-parameters-test
+  (testing "POST /api/card/:id/query with parameters with default values"
+    (mt/with-temp
+      [:model/Card card {:dataset_query
+                         {:database (mt/id)
+                          :type     :native
+                          :native   {:query         "SELECT count(*) AS Count FROM venues where {{venue_id}}"
+                                     :template-tags {"venue_id" {:dimension    [:field (mt/id :venues :id) nil],
+                                                                 :display-name "Venue ID",
+                                                                 :id           "_VENUE_ID_",
+                                                                 :name         "venue_id",
+                                                                 :required     false,
+                                                                 :default      1
+                                                                 :type         :dimension,
+                                                                 :widget-type  :id}}}}}]
+      (let [request (fn [body]
+                      (mt/user-http-request :rasta :post 202 (format "card/%d/query" (:id card)) body))]
+        (testing "the default can be overridden"
+          (is (= [[1]]
+                 (mt/rows (request {:parameters [{:id    "_VENUE_ID_"
+                                                  :name  "venue_id"
+                                                  :slug  "venue_id"
+                                                  :type  "number"
+                                                  :value 2}]})))))
+        (testing "the default should apply if no param value is provided"
+          (is (= [[1]]
+                 (mt/rows (request {:parameters []}))))
```

Comment:
What if parameter is passed but its value is unset?

Reply:
`value: null` and omitting the `value` key are equivalent.

Reply:
Got it, thanks.
Don't we have a convention somewhere that passing `nil` as a value (`null` in json) means "we deliberately want the value to be empty" while making it unset (`undefined` or no property in json) means "take default"?

Reply:
I'm pretty sure that was the behaviour before https://github.com/metabase/metabase/pull/31891, but now it's like this. Having a semantic difference between `value: null` and omitting the `value` key is asking for trouble, especially in Clojure where most functions will treat them the same.